### PR TITLE
Add rename button in all projects page

### PR DESCRIPTION
**Pull request by sae220**
splitted from https://github.com/microsoft/pxt/pull/10016

### CODE REVIEW

File: webapp/src/workspace.ts [hunk 1]
```diff
@@ -763,6 +763,21 @@ export function installAsync(h0: InstallHeader, text: ScriptText, dontOverwriteI
...
+    let newHdr = U.flatClone(h)
+
+    const dupText = U.flatClone(text);
+    newHdr.name = newName;
+    const cfg = JSON.parse(text[pxt.CONFIG_NAME]) as pxt.PackageConfig;
+    cfg.name = newHdr.name;
+    dupText[pxt.CONFIG_NAME] = pxt.Package.stringifyConfig(cfg);
+
+    return importAsync(newHdr, dupText)
+        .then(() => newHdr)
```
Comment (riknoll) [inline webapp/src/workspace.ts:778]:
```suggestion
    await importAsync(newHdr, dupText);
    return newHdr;
```

File: webapp/src/scriptmanager.tsx [hunk 2]
```diff
@@ -188,10 +189,31 @@ export class ScriptManagerDialog extends data.Component<ScriptManagerDialogProps
         this.props.parent.openNewTab(header, true);
     }
 
-    handleDuplicate() {
-        pxt.tickEvent("scriptmanager.dup", undefined, { interactiveConsent: true });
+    async handleRename() {
```
Comment (riknoll) [inline webapp/src/scriptmanager.tsx:192]:
```suggestion
    async handleRename() {
        pxt.tickEvent("scriptmanager.rename", undefined, { interactiveConsent: true });
```

**Review by riknoll — CHANGES_REQUESTED** [review]

**Comment by sae220** [comment]
@riknoll I don't know how pxt.tickEvent function works. Could you let me know briefly?

**Review by riknoll — APPROVED** [review]

**Comment by riknoll** [comment]
@sae220 LGTM! lint is failing because of trailing whitespace. you can run the lint locally with `gulp lint`

**Comment by riknoll** [comment]
@sae220 we don't have a docker container or anything like that, but we have people who work on Windows, Linux, and MacOS on the team so they should all work fine! I personally use Ubuntu on WSL, and that's what I would recommend if you're on Windows

**Comment by sae220** [comment]
@riknoll 
OK. I'll use Ubuntu on WSL.

Can I cherry pick this for stable9.3? As I said before, we can't rename projects in pxt-minecraft now.

**Comment by riknoll** [comment]
please do! tag me in the PR

**Pull request merged**